### ISSUE TITLE
Are We Upstream Yet?

### ISSUE THREAD

**Issue by janhohenheim** [Jondolf/avian#433]
Copying this [from a discussion we had a while ago on Discord](https://discord.com/channels/691052431525675048/1124043933886976171/1254029309216161802).
While Avian is currently not ready for upstreaming into Bevy, I want the information on *why* that is the case to be easily accessible, so I'm making this issue :)

Per @Jondolf:

- [ ] Performance parity with Rapier
	- [ ] Simulation islands
	- [ ] Better broad phase
	- [ ] Graph colouring for the solver
	- [ ] SIMD acceleration for the solver
- [ ] Feature parity with Rapier
	- idk what's missing
- [ ] Remove dependency on Parry
- [ ] Remove 2D / 3D dimensionality split

**Comment by alice-i-cecile** [comment]
Examples / character controllers for the most common genres would also be extremely useful. Namely:

- [ ] 2D platformer
- [ ] 3D platformer
- [ ] Top-down ARPG
- [ ] FPS

**Comment by janhohenheim** [comment]
@alice-i-cecile since character controllers are a very wide and complex topic with a lot of tradeoffs and discussions to be had, I would personally prefer upstreaming a physics engine first and then later talking about first-party character controllers in follow-ups. I agree that there should be examples for these things, but I would not block on a character controller being shipped with the engine.

**Comment by alice-i-cecile** [comment]
Yep, I don't think we should ship a first-party character controller in the library. Because they're full of tradeoffs and heavily customized, I tend to find that both character and camera controllers are better as vendored code rather than dependencies.

The motivation behind my suggestion is that a) that the crate can comfortably handle all of these common use cases and b) that there's a good place for beginners to start when tackling these use cases
